tools/mo/openvino/tools/mo/utils/telemetry_utils.py

```diff
@@ -22,13 +22,41 @@
 
 
 def init_mo_telemetry(app_name='Model Optimizer'):
-    return tm.Telemetry(tid=get_tid(),
-                        app_name=app_name,
-                        app_version=get_rt_version(),
-                        backend='ga4',
-                        enable_opt_in_dialog=False,
-                        disable_in_ci=True
-                        )
+    return init_telemetry_class(tid=get_tid(),
+                                app_name=app_name,
+                                app_version=get_rt_version(),
+                                backend='ga4',
+                                enable_opt_in_dialog=False,
+                                disable_in_ci=True
+                                )
+
+
+def init_telemetry_class(tid,
+                         app_name,
+                         app_version,
+                         backend,
+                         enable_opt_in_dialog,
+                         disable_in_ci):
+    # Init telemetry class
+    telemetry = tm.Telemetry(tid=tid,
+                             app_name=app_name,
+                             app_version=app_version,
+                             backend=backend,
+                             enable_opt_in_dialog=enable_opt_in_dialog,
+                             disable_in_ci=disable_in_ci)
+
+    # Telemetry is a singleton class and if it was already initialized in another tool
+    # some parameters will be incorrect, including app_name.
+    # In this case we need to force reinitialisation of telemetry.
+    if hasattr(telemetry, "backend") and telemetry.backend.app_name != app_name:
+        telemetry.init(tid=tid,
+                       app_name=app_name,
+                       app_version=app_version,
+                       backend=backend,
+                       enable_opt_in_dialog=enable_opt_in_dialog,
+                       disable_in_ci=disable_in_ci)
```

Comment:
Am I correct that both tools exist in the same process that is way they have the common telemetry object?
Also, if we re-initialize it, can it be a case that it starts to report incorrect `app_name` for the primary tool?

Reply:
> Am I correct that both tools exist in the same process that is way they have the common telemetry object?

Yes, it is correct.

> Also, if we re-initialize it, can it be a case that it starts to report incorrect `app_name` for the primary tool?

Yes, it may happen. But I expect all tools will use this logic with reinitialization to prevent that to happen.

Reply:
Discussed offline and agreed to make changes in Telemetry API. Namely, `send` method should have additional parameters for `app_name` and `version`, deprecate the current one and make tools to transition to new `send` method